tests/test_grammar.py

```diff
@@ -1,304 +1,437 @@
 from __future__ import absolute_import
+import re
 
 import os
-from unittest import TestCase, main
+from unittest import TestCase, main, SkipTest
 
 from lark import Lark, Token, Tree, ParseError, UnexpectedInput
 from lark.load_grammar import GrammarError, GRAMMAR_ERRORS, find_grammar_errors, list_grammar_imports
 from lark.load_grammar import FromPackageLoader
+from lark.lark_validator import LarkValidator
+
+
+__all__ = []
+
+class LarkDotLark:
+    def __init__(self, grammar, **kwargs):
+        options = {}
+        options.update(kwargs)
+        if "start" in options and options["start"] != "start":
+            # We're not going to parse with the parser, so just override it.
+            options["start"] = "start"
+        options["propagate_positions"] = True
+        lark_parser = Lark.open_from_package("lark", "grammars/lark.lark", **options)
+        tree = lark_parser.parse(grammar)
+        LarkValidator.validate(tree)
+
+    def parse(self, text: str, start=None, on_error=None):
```

Comment:
Looks to me like it would be better if the tests that try to call `parse()` wouldn't be inside `_make_tests`, since their dot-lark variation will be skipped anyway.

Reply:
I had some hope when I wrote that stuff that I could solve the "can't call parse()" problem.  I made several attempts, all failed.  If you want me to move them out, I will.

Reply:
If it's not too much trouble, yes.